lib/team.rb

```diff
@@ -63,12 +62,41 @@ def average_goals_home
   # end
 
   def total_winning_percentage
-    all_games_played = Game.all.find_all do |game|
-      game.home_team_id == @team_id || game.away_team_id == @team_id
-    end
     all_games_won = all_games_played.find_all do |game|
       game.winner == @team_id
     end
     (all_games_won.length.to_f / all_games_played.length).round(2)
   end
+
+  def home_win_percentage
+    all_home_games = all_games_played.find_all do |game|
+      game.home_team_id == @team_id
+    end
+    (home_games_won.length.to_f / all_home_games.length).round(2)
+  end
+
+  def away_win_percentage
+    all_away_games = all_games_played.find_all do |game|
+      game.away_team_id == @team_id
+    end
+    (away_games_won.length.to_f / all_away_games.length).round(2)
+  end
+
+  def all_games_played
+    all_games_played = Game.all.find_all do |game|
+      game.home_team_id == @team_id || game.away_team_id == @team_id
+    end
+  end
+
```

Comment:
Now that we have an all_games_played we can probably refactor home_games_won and away_games_won to speed up our testing